ArduPlane/Attitude.cpp

```diff
@@ -564,6 +564,11 @@ void Plane::calc_nav_roll()
             millis() - plane.guided_state.last_forced_rpy_ms.x < 3000) {
         commanded_roll = plane.guided_state.forced_rpy_cd.x;
     }
+    else if (control_mode == LOITER &&
+         g2.soaring_controller.is_active() &&
+         g2.soaring_controller.is_controlling_roll()) {
+        commanded_roll = g2.soaring_controller.get_roll_cmd();
```

Comment:
I'm reeeeeelly not a fan of people overriding roll angles like this. It tells me that they've tuned it for a single aircraft and never considered any other aircraft. The Guided one is there for an external system to control the aircraft and the Avoid one is there for evasive maneuvers. This would be MUCH better handled with the soaring system either switching to LOITER to setting a Guided point that expires

Reply:
The existing soaring mode does just that, uses the built-in navigation controller by setting a waypoint. Unfortunately the built-in controllers aren't optimised for soaring. This feature is essentially adding a new navigation controller that uses knowledge of the estimated thermal location to select a flightpath, and is tuned using the parameters RLLRAT, RLLDMP, I_MOM etc. I do think that there are too many parameters and this will make it difficult to tune, so I hope to reduce it by making some assumptions and reusing more parameters from the roll controller.